### ISSUE TITLE
Use this project with recent GPU ?

### ISSUE THREAD

**Issue by antoinersw** [sicxu/Deep3DFaceRecon_pytorch#169]
Hello I have a 4080 (sm_89), according to the warning log CUDA is not compatible. 

Is there a way to use a more recent version of CUDA/pytorch ?
![image](https://github.com/sicxu/Deep3DFaceRecon_pytorch/assets/57794470/37eedf85-1312-4959-880a-e6bbb0fe24a1)

Regards

**Comment by vladgohn** [comment]
same problem, no answers

**Comment by sebaturen** [comment]
In our case, we've opted for switching to a GPU compatible with CUDA11, specifically with version sm_75. While this approach is functional, it's not an optimal solution. Installing and configuring CUDA with a specific version can be challenging.
* we switch to GTX 20xx

**Comment by vladgohn** [comment]
I would generally like to run the project somehow, even on a CPU, just to make it work even slowly. do I have a chance to do this?

**Comment by sebaturen** [comment]
I don't think you can run it on a CPU; TensorFlow requires a GPU. For now, I managed to run the project by changing the GPU, but I'm having issues generating the landmarks.

**Comment by OchotonaPrinceps** [comment]
I managed to run it on an RTX4060.
![screenshot](https://github.com/sicxu/Deep3DFaceRecon_pytorch/assets/156045467/997bd58d-9f6c-40ba-bc67-17aac794c8bb)

It took me a couple hours, but I am happy with the results.

**Comment by imxtx** [comment]
> I managed to run it on an RTX4060. ![screenshot](https://private-user-images.githubusercontent.com/156045467/305566323-997bd58d-9f6c-40ba-bc67-17aac794c8bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1OTgzNTYsIm5iZiI6MTcwODU5ODA1NiwicGF0aCI6Ii8xNTYwNDU0NjcvMzA1NTY2MzIzLTk5N2JkNThkLTlmNmMtNDBiYS1iYzY3LTE3YWFjNzk0YzhiYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyMlQxMDM0MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMmE0Mjc3OWFiYzMxNGI1MWVhMWE1MDQzZmQ2Mjg5ODhhMjg3YWNlZjNhYzlhY2E4NjczYWU1OTY5NTQyMTNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.9jGPoPnHg2IZ1kI8oMQTXkuBeKv506dM8MS84DnWb1A)
> 
> It took me a couple hours to set up, but I am happy with the results.

Nice! Can you tell us how did you solve this problem?

**Comment by toseefkhan403** [comment]
> I managed to run it on an RTX4060. ![screenshot](https://github.com/sicxu/Deep3DFaceRecon_pytorch/assets/156045467/997bd58d-9f6c-40ba-bc67-17aac794c8bb)
> 
> It took me a couple hours to set up, but I am happy with the results.

Hey! I'm stuck on the same problem, can you tell us how you got it working? Also, how long does the process usually take?

TIA 🐱

**Comment by urlan** [comment]
> I managed to run it on an RTX4060. ![screenshot](https://github.com/sicxu/Deep3DFaceRecon_pytorch/assets/156045467/997bd58d-9f6c-40ba-bc67-17aac794c8bb)
> 
> It took me a couple hours to set up, but I am happy with the results.

Can you show us all modifications you have made? This project is old, so I'm quite sure it'll have a lot of problems to work properly.

**Comment by LauraCrx** [comment]
> Can you show us all modifications you have made? This project is old, so I'm quite sure it'll have a lot of problems to work properly.

you have to make a conda env with a cuda-toolkit compatible with your card and then install a version of pytorch3d, torch and all other packages that are compatible with the cuda-toolkit your choice.

a exemples from another projetc:
pytorch3d-0.7.6 pytorch==2.3.0 torchvision==0.18.0 torchaudio==2.3.0 pytorch-cuda=**12.1**
